crates/katana/executor/src/implementation/blockifier/utils.rs

```diff
@@ -568,42 +549,45 @@ fn starknet_api_ethaddr_to_felt(
     bytes[12..32].copy_from_slice(value.0.as_bytes());
-    let stark_felt = katana_cairo::starknet_api::hash::StarkFelt::new(bytes)
-        .expect("valid slice for stark felt");
-    to_felt(stark_felt)
+    FieldElement::from_bytes_be(&bytes)
 }
 
 pub fn to_exec_info(exec_info: TransactionExecutionInfo) -> TxExecInfo {
     TxExecInfo {
         validate_call_info: exec_info.validate_call_info.map(to_call_info),
         execute_call_info: exec_info.execute_call_info.map(to_call_info),
         fee_transfer_call_info: exec_info.fee_transfer_call_info.map(to_call_info),
-        actual_fee: exec_info.actual_fee.0,
-        actual_resources: exec_info
-            .actual_resources
-            .0
-            .into_iter()
-            .map(|(k, v)| (k, v as u64))
-            .collect(),
+        actual_fee: exec_info.transaction_receipt.fee.0,
         revert_error: exec_info.revert_error.clone(),
+        actual_resources: TxResources {
+            vm_resources: exec_info.transaction_receipt.resources.vm_resources,
+            n_reverted_steps: exec_info.transaction_receipt.resources.n_reverted_steps,
+            data_availability: L1Gas {
+                l1_gas: exec_info.transaction_receipt.da_gas.l1_data_gas,
+                l1_data_gas: exec_info.transaction_receipt.da_gas.l1_data_gas,
+            },
+            total_gas_consumed: L1Gas {
+                l1_gas: exec_info.transaction_receipt.gas.l1_data_gas,
+                l1_data_gas: exec_info.transaction_receipt.gas.l1_data_gas,
+            },
+        },
     }
 }
 
 fn to_call_info(call: CallInfo) -> trace::CallInfo {
     let contract_address = to_address(call.call.storage_address);
     let caller_address = to_address(call.call.caller_address);
     let code_address = call.call.code_address.map(to_address);
-    let class_hash = call.call.class_hash.map(|a| to_felt(a.0));
-    let entry_point_selector = to_felt(call.call.entry_point_selector.0);
-    let calldata = call.call.calldata.0.iter().map(|f| to_felt(*f)).collect();
-    let retdata = call.execution.retdata.0.into_iter().map(to_felt).collect();
-
-    let builtin_counter = call.resources.builtin_instance_counter;
-    let execution_resources = trace::ExecutionResources {
-        n_steps: call.resources.n_steps as u64,
-        n_memory_holes: call.resources.n_memory_holes as u64,
-        builtin_instance_counter: builtin_counter.into_iter().map(|(k, v)| (k, v as u64)).collect(),
-    };
+    let class_hash = call.call.class_hash.map(|a| a.0);
+    let entry_point_selector = call.call.entry_point_selector.0;
+    let calldata = call.call.calldata.0.as_ref().clone();
+    let retdata = call.execution.retdata.0;
+
+    // let execution_resources = trace::ExecutionResources {
+    //     n_steps: call.resources.n_steps,
+    //     n_memory_holes: call.resources.n_memory_holes as u64,
+    //     builtin_instance_counter: call.resources.builtin_instance_counter,
+    // };
```

Comment:
We still need that in a second iteration with the new `TxResources`? Or can be removed?

Reply:
ah this should be removed.